robottelo/hosts.py

```diff
@@ -883,12 +883,17 @@ def get(self, remote_path, local_path=None):
         """Get a remote file from the broker virtual machine."""
         self.session.sftp_read(source=remote_path, destination=local_path)
 
-    def put(self, local_path, remote_path=None):
+    def put(self, local_path, remote_path=None, temp_file=False):
         """Put a local file to the broker virtual machine.
         If local_path is a manifest object, write its contents to a temporary file
         then continue with the upload.
         """
-        if 'utils.manifest' in str(local_path):
+        if temp_file:
+            with NamedTemporaryFile(dir=robottelo_tmp_dir) as content_file:
+                content_file.write(str.encode(local_path))
```

Comment:
what is going on here? how is this different from the `utils.manifest` case?

Reply:
I honestly don't know how the path is being passed for the case where the local path contains `utils.manifest`. For my instance, I can at least say you aren't actually passing a path but rather a string. Whereas the manifest case does some sort of `read()` function. Maybe @JacobCallahan or @synkd could give us some insight here?

Reply:
The path is being passed by the `upload_manifest` helper [1], specifically in cases where we're uploading a manifest via CLI, rather than imported directly via API. In those cases, we need to pass the path of the manifest file needs to be `put` onto the Satellite before being imported with hammer. The third bullet point in this comment [2] describes these scenarios.

I think that, in cases where Manifester times out and we fail over to cloned manifests, the contents of the cloned manifest object need to be written to a file before transferring that file to the Satellite, whereas with Manifester, the file would already be present on the local file system. 

I hope this provides some measure of clarity. It is admittedly rather confusing.

[1] https://github.com/SatelliteQE/robottelo/blob/bba4931f4aec31af9288f58e0bff3cbabc9cc6ca/robottelo/host_helpers/satellite_mixins.py#L137
[2] https://github.com/SatelliteQE/robottelo/pull/12605#issuecomment-1718143962